src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -24,13 +24,13 @@
 class EmailValidator extends ConstraintValidator
 {
     /**
-     * @var bool
+     * @var string
      */
-    private $isStrict;
+    private $mode;
 
-    public function __construct($strict = false)
+    public function __construct($mode = 'html5')
```

Comment:
we need to make the change in a BC friendly way, so I think we cannot change the default behavior
true/false should still be handled as before as much as this makes sense

Reply:
Something along these lines? 

```php
    /**
     * @var string
     */
    private $mode;

    public function __construct($strict = false, $mode = 'html5')
    {
        $this->mode = $mode;

        if($strict === true) {
            $this->mode = Email::VALIDATION_MODE_STRICT;
        }
    }
```

Reply:
No, we should have only 1 argument (as we don't want to keep a first unused argument in the future API)